unity_jobs_simulator/Assets/Scripts/Robot_Controller.cs

```diff
@@ -0,0 +1,273 @@
+        
+        if(SamePos(CurrentJointAngles, TargetJointAngles)) // If we are at the preset target, FixedUpdate shouldn't run so we return nothing
+        {
+            if(CheckAngleError)
+            {
+                CheckAngleErrors();
+            }
+            return;
+        }
+
+        if(!(SamePos(_currentDriveTarget, _enroute))) // If the route has changed, we need to restart the motion
+        {
+            _enroute = new List<float>(TargetJointAngles);
+            _incAngles = new List<float>(_zeroList);
+        }
+
+        if(SamePos(_incAngles, _zeroList)) // If we don't have increments for the angles, then we aren't going anywhere and need to initialize these increments
+        {
+            _maxVelRad = MaxVelocityDeg * Mathf.Deg2Rad; // This updates the value of _maxVelRad if the input MaxVelocityDeg is changed during motion
+            GetIncrements(); // This calculates the values for increments given the target angles, the current positions, and the maximum velocity
+            SetIncAngles(); // This calculates the incremental angles given the increments and the target angles
+            _enroute = new List<float>(TargetJointAngles); // We set our enroute position as this is where our robot movement will be moving to
+            _nextAngles = AddLists(_currentDriveTarget, _incAngles); // This increments the angles by one increment
+        }
+
+        if(SamePos(_nextAngles, TargetJointAngles)) // If the next Angle is the final target position, then this is the final increment
+        {
+            AdjustRobotAngles(TargetJointAngles, 0f); // We set the final angle to the final target, and we set the joint the velocities to zero
+            _incAngles = new List<float>(_zeroList); // We've reached the final location so we do not want to increment the angles anymore so we set this to zero
+        }
+        else // Otherwise, the robot is incrementing the angles to the next incremental position
+        {
+            AdjustRobotAngles(_nextAngles, _maxVelRad); // We adjust the robot to the next position
+            _nextAngles = AddLists(_nextAngles, _incAngles); // We increment the nextAngle list up by the incremental angles
+        }
+    }
+
+    public void UpdateCurrentDrivePos()
+    {
+        _currentDriveTarget = new List<float>(); // Creates a new list to store the current Drive target positions in
+        for(int i = 0; i < _joints.Count ; i++) // Runs through each Joint to check its position
+        {
+            ArticulationBody joint = _joints[i];
+            float temp = joint.xDrive.target;
+            _currentDriveTarget.Add(temp);
+        }
+
+        CurrentJointAngles = new List<float>(); 
+        for(int i = 0; i < _joints.Count ; i++) 
+        {
+            ArticulationBody joint = _joints[i];
+            float temp = joint.jointPosition[0];
+            CurrentJointAngles.Add(temp * Mathf.Rad2Deg);
+        }
+    }
+
+    public void GetIncrements()
```

Comment:
This is assuming the time taken for different joints to complete motion is different. Previously we spoke about having each joint finish its total movement at the same time. 

We can leave this as is for now in the interest of time (might be a more complicated change?) but can you add in a TODO to adjust this logic later on?